src/App.js

```diff
@@ -1,16 +1,132 @@
 import React from "react";
 import logo from "./logo.png";
 import "./App.css";
+import axios from "axios";
 
 class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      cityInputValue: "",
+      city: "",
+      weather: {},
+      weatherIcon: "",
+      coordinates: { lat: "", lon: "" },
+      forecastWeather: [],
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ cityInputValue: event.target.value });
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    axios
+      .get(
+        `https://api.openweathermap.org/geo/1.0/direct?q=${this.state.cityInputValue}&limit=1&appid=b19b32221939a66675b01f5452af913c`
+      )
+      // City geo data is in response.data[0]
+      // Arrow functions with no curly braces return value after arrow
+      .then((response) => {
+        this.setState({
+          coordinates: { lat: response.data[0].lat, lon: response.data[0].lon },
+        });
+        return response.data[0];
+      })
+      .then((cityGeoData) =>
+        axios.get(
+          `https://api.openweathermap.org/data/2.5/weather?lat=${cityGeoData.lat}&lon=${cityGeoData.lon}&appid=b19b32221939a66675b01f5452af913c&units=metric`
+        )
+      )
+      .then((response) => {
+        const { data: weatherData } = response;
+        console.log(this.state.coordinates);
+        this.setState({
+          city: this.state.cityInputValue,
+          weather: weatherData.main,
+          weatherIcon: weatherData.weather[0],
+          cityInputValue: "",
+        });
+      })
+      .then(() =>
+        axios.get(
+          `https://api.openweathermap.org/data/2.5/forecast?lat=${this.state.coordinates.lat}&lon=${this.state.coordinates.lon}&appid=b19b32221939a66675b01f5452af913c&units=metric`
+        )
+      )
+      .then((response) => {
+        console.log(response.data.list);
+        this.setState({ forecastWeather: response.data.list });
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };
+
   render() {
+    console.log(this.state.forecastWeather);
+    const forecastedWeather = this.state.forecastWeather;
```

Comment:
No problem here, but we can also use destructuring! 
```suggestion
    const { forecastedWeather } = this.state;
```

Reply:
You could do that then for all state variables used below, and then easily access them. 
```suggestion
    const { forecastedWeather, city, weatherIcon, weather } = this.state;
    const { temp, feels_like, temp_min, temp_max, humidity } = weather;
```